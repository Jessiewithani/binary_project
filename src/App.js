import React, { Component } from 'react';
import './App.css';
import InfoCardContainer from './Components/InfoCardContainer/InfoCardContainer'

class App extends Component {
  constructor() {
    super()
    this.state = {
      characterData: []
    }
  }
  componentDidMount() {
    fetch('https://rickandmortyapi.com/api/character')
      .then(response => response.json())
      .then(data => {
        return data.results.map(data => data)
      })
      .then(characters => this.setState({characterData: characters}))
  }
  render() {
    console.log('CHARACTERS DATA', this.state.characterData)
    return(
      <main className="App">
        <h1>hi app you fucker</h1>
        <InfoCardContainer/>
      </main>
    )
  }
}

export default App;
