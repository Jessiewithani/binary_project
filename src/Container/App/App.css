.App {
  text-align: center;
  margin-left: 3.2em;
  margin: 0;
}

@font-face {
  font-family: 'Get Schwifty';
  font-weight: normal;
  font-style: normal;
  src: url('../../Fonts/GetSchwifty-Regular.woff') format(‘woff’)
}
