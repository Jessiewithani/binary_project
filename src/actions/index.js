export const saveCharacters = (characters) => ({
    type: 'SAVE_CHARACTERS',
    characters
})

export const saveFavorites = (favorites) => ({
    type: 'SAVE_FAVORITES',
    favorites
})

export const removeFavorites = (id) => ({
    type: 'REMOVE_FAVORITES',
    id
})

export const searchCharacter = (name) => ({
    type: 'SEARCH_CHARACTERS',
    name
})