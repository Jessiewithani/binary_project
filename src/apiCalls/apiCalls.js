export const getCharacters = async (url = 'https://rickandmortyapi.com/api/character') => {
    const response = await fetch(url)
    if(!response.ok) {
        throw Error('could not get characters')
    } else {
        let characters = await response.json()
        return {...characters}
    }
}

