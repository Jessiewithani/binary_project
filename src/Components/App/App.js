import React, { Component } from 'react';
import '../App/App.css';
import CharacterCardContainer from '../CharacterCardContainer/CharacterCardContainer';
import { getCharacters } from '../../apiCalls/apiCalls'

class App extends Component {
  constructor() {
    super()
    this.state = {
      characterData: []
    }
  }
  componentDidMount() {
    getCharacters()
      .then(characters => this.setState({characterData: characters}))
  }
  render() {
    console.log('CHARACTERS DATA', this.state.characterData)
    return(
      <main className="App">
        <h1>hi app you fucker</h1>
        <CharacterCardContainer characterData={this.state.characterData} />
      </main>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  
})

export default App;
