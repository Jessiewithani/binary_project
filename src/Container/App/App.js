import React, { Component } from 'react';
import CharacterCardContainer from '../CharacterCardContainer/CharacterCardContainer';
import LandingPage from '../../Components/LandingPage/LandingPage';
import Pagination from '../Pagination/Pagination';
import { Route } from 'react-router-dom'
import { connect } from 'react-redux';
import { getCharacters } from '../../apiCalls/apiCalls'
import { saveCharacters } from '../../actions'
import '../../Container/App/App.css';

export class App extends Component {
  constructor() {
    super() 
    this.state = {
      next: '',
      prev: ''
    }
  }
  componentDidMount() {
    getCharacters()
    .then(characters => {
      this.setState({next: characters.info.next, prev: characters.info.prev})
      this.props.allCharacters(characters)
    });
  }

  nextPage = (url) => {
    getCharacters(url)
    .then(characters => {
      this.setState({next: characters.info.next, prev: characters.info.prev})
      this.props.allCharacters(characters)
    });
}

render() {
    return(
      <main className="App">
        <Route exact path='/' component={LandingPage} />
        <Route exact path='/characters' component={CharacterCardContainer} />
        <Pagination nextPage={this.nextPage} next={this.state.next} prev={this.state.prev}/>
      </main>
    )
  }
} 

export const mapDispatchToProps = (dispatch) => ({
  allCharacters: (characters) => dispatch(saveCharacters(characters)),
})


export default connect(null, mapDispatchToProps)(App)
