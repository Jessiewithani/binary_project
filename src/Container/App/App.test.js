import React from 'react';
import { App, mapDispatchToProps } from '../App/App';
import { saveCharacters } from '../../actions'
import { getCharacters } from '../../apiCalls/apiCalls'
import { shallow } from 'enzyme';

jest.mock('../../apiCalls/apiCalls')

describe('App', () => {
  beforeEach(() => {
    getCharacters.mockImplementation(() => {
      return Promise.resolve([{
        name: 'Morty Smith',
        status: 'Alive',
        species: 'Human',
        type: '',
        gender: 'Male',
        location: 'Earch (Replaement Dimension'
    }]);
  })
})
  it('should match the snapshot with the correct data passing through', () => {
    const wrapper = shallow(<App/>)
    expect(wrapper).toMatchSnapshot()
  })
  it('should retrieve characters after mounting', () => {
    shallow(<App/>);
    expect(getCharacters).toHaveBeenCalled()
  })
})

describe('mapDispatchToProps in App', () => {
  it('calls dispatch with a saveCharacters action when allCharacters is called', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = saveCharacters([{
      name: 'Morty Smith',
      status: 'Alive',
      species: 'Human',
      type: '',
      gender: 'Male',
      location: 'Earch (Replaement Dimension'
    }, {
      name: 'Big Head Morty',
      status: 'unknown',
      species: 'Human',
      type: 'Humam with giant head',
      gender: 'Male',
      location: 'Citadel of Ricks'
      }]);

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.allCharacters([{
      name: 'Morty Smith',
      status: 'Alive',
      species: 'Human',
      type: '',
      gender: 'Male',
      location: 'Earch (Replaement Dimension'
    } , {
        name: 'Big Head Morty',
        status: 'unknown',
        species: 'Human',
        type: 'Humam with giant head',
        gender: 'Male',
        location: 'Citadel of Ricks'
      }
    ])

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})
