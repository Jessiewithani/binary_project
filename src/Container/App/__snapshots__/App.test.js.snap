// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with the correct data passing through 1`] = `ShallowWrapper {}`;
